be/src/exec/spill/log_block_manager.cpp

```diff
@@ -45,6 +45,8 @@ class LogBlockContainer {
     ~LogBlockContainer() {
         TRACE_SPILL_LOG << "delete spill container file: " << path();
         WARN_IF_ERROR(_dir->fs()->delete_file(path()), fmt::format("cannot delete spill container file: {}", path()));
+        // try to delete related dir, only the last one can success, we ignore the error
+        (void)(_dir->fs()->delete_dir(parent_path()));
```

Comment:
this logical should be in DirManager?

Reply:
> this logical should be in DirManager?

nope, DirManager does not care about specific queries, it is only responsible for allocating writable directories.
LogBlockContainer is bound to the query, we need to handle the creation and deletion of the directory during its life cycle